src/errors/abort-error.js

```diff
@@ -4,7 +4,9 @@ import {FetchBaseError} from './base.js';
  * AbortError interface for cancelled requests
  */
 export class AbortError extends FetchBaseError {
-	constructor(message, type = 'aborted') {
```

Comment:
Is this a breaking change?

`AbortError` is exported publicly from the package...

Reply:
Yeah, wasn't aware the constructor was actually exported. It just seemed odd that the `type` would be explicitly given, i.e. what else could it be.

We basically need to thread the `reason` into the thrown `AbortError`. To not break its constructor, the only sane option would be to extend it:
```js
  // no change to existing `AbortError`
  class AbortWithReasonError extends AbortError {
    constructor(message, reason) {
      super(message);
      this.reason = reason;
    }
  }
```